Automattic/wp-calypso
# Onboarding: show the draft post modal when the user choose write intent

#### Changes proposed in this Pull Request

* Implement `<DraftPostModal />` component with the following behaviors:
  * Click [X] to verify the modal is closed and stay in the editor
  * Click [Start writing] to verify the modal closed and stay in the editor
  * Click [I'm not ready] to verify you go to My Home
* When the user selects the write intent and  finishes the setup-site flow, it will set `wpcom_signup_complete_show_draft_post_modal` to session storage and redirect to the editor. When landing on the editor, it will read that value and cleanup the session storage. Next, using this value to set `showDraftPostModal` in query args, and then the `wpcom-block-editor-nux` can retrieve this value and decide to show the `<DraftPostModal />` or `<LaunchWpcomWelcomeTour />`. Besides, as the value is cleaned up immediately after reading, the `<DraftPostModal />` will only show at the first time. When the user lands on the editor next time, it will show the `<LaunchWpcomWelcomeTour />`.
* The reason I use `sessionStorage` is that I just want to keep the URL looks clean and we can easily cleanup the value without any redirecting.

https://user-images.githubusercontent.com/13596067/136190238-deac32e1-899f-4402-9df5-51db59800244.mov

![image](https://user-images.githubusercontent.com/13596067/136183188-a83105c8-d2be-47ab-80ec-66ff88915d28.png)

![image](https://user-images.githubusercontent.com/13596067/136183932-644e17cc-7789-4825-8f2b-6d4901212e1f.png)


#### Testing instructions



* Sandbox your API and ensure your siteSlug domain point to your sandbox
* Open another terminal and run `cd apps/editing-toolkit && yarn dev --sync`
* Go to `/start/setup-site?siteSlug=arthurtestonboarding.wordpress.com&flags=signup/hero-flow`
* Select write intent and you'll redirect to post editor
* Verify write your post modal show
* Verify the following behaviors one by one (note that when you finish one, you need to repeat to choose the intent)
  * Click [X] to verify the modal is closed and stay in the editor
  * Click [Start writing] to verify the modal closed and stay in the editor
  * Click [I'm not ready] to verify you go to My Home
* Refresh the page and verify the write your post modal not show but welcome tour shows



Related to https://github.com/Automattic/wp-calypso/issues/55603
